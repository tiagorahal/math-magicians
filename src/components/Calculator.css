.calculator {
  max-width: 400px;
  margin: 0 auto;
}

.display {
  height: 2rem;
  padding: 1rem;
  font-size: 2rem;
  text-align: end;
  background-color: #888;
  color: #fff;
}

.keyboard {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.key {
  padding: 1rem;
  border: 1px solid #ccc;
  text-align: center;
  background-color: #eee;
  user-select: none;
}

.orange {
  background-color: #ffa500;
}

.zero {
  grid-column-start: 1;
  grid-column-end: span 2;
}
