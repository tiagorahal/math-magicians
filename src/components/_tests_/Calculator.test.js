import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Calculator from '../calculator';

it('matches the snapshot', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders AC key element', () => {
  render(<Calculator />);
  const acKeyElement = screen.getByText(/AC/i);

  expect(acKeyElement).toBeInTheDocument();
});

it('recognizes clicks on numeric buttons', () => {
  render(<Calculator />);
  const Key2Element = screen.getByText(/2/i);
  const Key4Element = screen.getByText(/4/i);

  fireEvent.click(Key4Element);
  fireEvent.click(Key2Element);

  const displayElement = screen.getByText(/42/i);

  expect(displayElement.textContent).toMatch(/42/i);
});

it('multiplies two numbers', () => {
  render(<Calculator />);
  const Key4Element = screen.getByText(/4/i);
  const KeyTimesElement = screen.getByText(/x/i);
  const KeyEqualsElement = screen.getByText(/=/i);

  fireEvent.click(Key4Element);
  fireEvent.click(KeyTimesElement);
  fireEvent.click(Key4Element);
  fireEvent.click(KeyEqualsElement);

  const displayElement = screen.getByText(/16/i);

  expect(displayElement.textContent).toMatch(/16/i);
});
