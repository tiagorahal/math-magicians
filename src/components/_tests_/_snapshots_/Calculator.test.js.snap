// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="calculator"
>
  <div
    className="display"
  >
      
  </div>
  <div
    className="keyboard"
  >
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="key orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="key orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="key orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="key orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="key zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="key"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="key orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;